CMakeLists.txt

```diff
@@ -95,6 +95,54 @@ else()
 	message(WARNING "Support only GCC, Clang, MSVC and AppleClang. Your compiler may or may not work.")
 endif()
 
+# Add flags for AVX instructions if available on the machine (and NETWORKIT_NATIVE is set to ON)
+if (NETWORKIT_NATIVE)
+
+	# Warning message to throw if AVX instructions are not available on host.
+	set(AVX_WARNING_MESSAGE "AVX instructions are not available on this machine")
+
+	# Check if avx512f is available on host. Store result in AVX_512f_TRUE.
+	function(check_avx512f CPUINFO)
+		STRING(REGEX REPLACE "^.*(avx512f).*$" "\\1" AVX_THERE ${CPUINFO})
+		STRING(COMPARE EQUAL "avx512f" "${AVX_THERE}" AVX_FOUND)
+		set(AVX_512f_TRUE ${AVX_FOUND} PARENT_SCOPE)
+	endfunction(check_avx512f)
+
+	# Check if avx512vl is available on host. Store result in AVX_512vl_TRUE.
+	function(check_avx512vl CPUINFO)
+		STRING(REGEX REPLACE "^.*(avx512vl).*$" "\\1" AVX_THERE ${CPUINFO})
+		STRING(COMPARE EQUAL "avx512vl" "${AVX_THERE}" AVX_FOUND)
+		set(AVX_512vl_TRUE ${AVX_FOUND} PARENT_SCOPE)
+	endfunction(check_avx512vl)
+
+	# Check if AVX instructions are supported on host. If yes, enable the required AVX flags.
+	function(set_avx_flags CPUINFO)
+		check_avx512f(${CPUINFO})
+		check_avx512vl(${CPUINFO})
+
+		if (AVX_512f_TRUE AND AVX_512vl_TRUE)
+			set(NETWORKIT_CXX_FLAGS "${NETWORKIT_CXX_FLAGS} -mavx -mavx512vl -mavx512f -DNETWORKIT_WITH_AVX")
```

Comment:
I wonder if the support for `AVX512` necessary here. You use 256bit-width variables (`__m256i`), which are also supported by `AVX2` ([Intrinsics Guide from Intel](__m256i)). This would lead to many more users/platforms able to use it. Does 512 vs 2 result in a speed-up?

Disclaimer:
Not an expert on this subject, but the common sense (at least on Skylake microarchitecture) is, that a speed-up in real-life applications is only expected if 512-bit width variables are used. And even then it is sublinear, since `AVX512` functions have a higher latency.